### DISCUSSION TITLE
How to add Sceneform Maintained to existing Android project?

### DISCUSSION THREAD

**Discussion by agGitHub** [sceneview/sceneform-android#227] [Q&A] [answered]
Hi,

I am trying to add Sceneform Maintained to existing Android project. But didn't manage so far...

What I did, as explained here:

1) Copy/paste the folders 'sceneform' 'core' and 'ux' from sample project to the root of my project

2) Add the following code to settings.gradle file:

```
include ':core', ':ux', ':sceneform'

// include ':core'
project(':core').projectDir=new File('my_root_folder/core')

// include ':ux'
project(':ux').projectDir=new File('my_root_folder/ux')
```

3) Add the following in dependencies of the build.gradle (app) file:
```
    api project(":ux")
    releaseImplementation "com.gorisse.thomas.sceneform:sceneform:1.20.2"
    debugApi project(":sceneform")
```

My project builds when I click "synch" but I have got many errors in my SceneFormActivity (where I have got all my Sceneform code) as the 'import com.google.ar.sceneform' tags generate errors (Android studio doesn't know where to find these files)

Note that the core/ux/sceneform folders are not displayed in the same way as in the sample project ; indeed, the folders icons are different (green circles in the sample project vs blue squares in my project). Also, nothing appears when I click on my core/ux/sceneform (can't open these folders, just icons that can be expanded). Also, when I go to 'Project structure' => 'Modules', and then I select one of the modules core/ux/sceneform, I have got the message 'Nothing to show. Please select an Android module" (in the sample project, I can have details about each module)

Any idea how to fix this problem? Or any detailed explanations to tell how to import Sceneform Maintained in an existing project?

Thanks !

**Comment by RGregat** [comment, marked as answer]
In general this should be enough

```
implementation "com.gorisse.thomas.sceneform:sceneform:1.20.2"
```

The first approach should work as well, make sure that the `sceneform-android` folder is in the root folder of your project. 

```
include ':app', ':core', ':ux'

project(':core').projectDir = new File('sceneform-android/core')
project(':ux').projectDir = new File('sceneform-android/ux')
```

Also use only the Dependency OR the direkt import of the source files not both. Also try to invalidate the cache, that can help!
Tomorrow I can take a look on one of my projects, there I have both ways implemented and I comment the Dependency or the source variant out in in, depending on my needs.
